### ISSUE TITLE
Ensure presence of claims

### ISSUE THREAD

**Issue by AlexPatrasco** [jwt/ruby-jwt#244]
Good day,
Is there a way to check for presence of particular claims when decoding JWT.

Sample scenario:
JWT is used as request signature & expiration flag. Exp claim cannot fail if there is no such claim in payload. Is there a way to ensure presence of said `exp` claim within payload?

**Comment by excpt** [comment]
Currently there is no option enforcing presence of a token. This is something that has to be implemented system specific. After decoding a token the payload is returned and it is possible to look for the presence of tokens.

Example:

```ruby
payload, header = JWT.decode token, secret, true, 'HS256'

payload.include?('exp')
# false
```

**Comment by excpt** [comment]
@AlexPatrasco Did you try implementing the suggested code?

**Comment by RootTJNII** [comment]
I'd also like to +1 this.  It would be nice to pass in an array of standard claims and treat the token as invalid if they're not present.  This will allow for cleaner server-side control of what constitutes a valid token for the application.

**Comment by rahulbajaj0509** [comment]
I would like to work on this one. Is it possible to assign this issue to myself?

**Comment by andyjdavis** [comment]
I have raised a PR with a possible solution to this. Feedback welcome. https://github.com/jwt/ruby-jwt/pull/430

**Comment by andyjdavis** [comment]
I think this issue can now be closed.

**Comment by andyjdavis** [comment]
This issue is resolved by changes available in version 2.3.0. https://github.com/jwt/ruby-jwt/releases/tag/v2.3.0

I don't have the ability to mark this issue as closed. Is someone else able to?
